targets/TARGET_Analog_Devices/TARGET_ADUCM4X50/TARGET_ADUCM4050/api/spi_api.c

```diff
@@ -0,0 +1,346 @@
+ *   3  |  1   1
+ * @endcode
+ */
+void spi_format(spi_t *obj, int bits, int mode, int slave)
+{
+    ADI_SPI_HANDLE  SPI_Handle;
+    ADI_SPI_RESULT  SPI_Return = ADI_SPI_SUCCESS;
+    bool_t          master;
+
+    master = !((bool_t)slave);
+    SPI_Handle = *obj->pSPI_Handle;
+    SPI_Return = adi_spi_SetMasterMode(SPI_Handle, master);
+    if (SPI_Return) {
+        obj->error = SPI_EVENT_ERROR;
+        return;
+    }
+}
+
+
+/** Set the SPI baud rate
+ *
+ * Actual frequency may differ from the desired frequency due to available dividers and bus clock
+ * Configures the SPI peripheral's baud rate
+ * @param[in,out] obj The SPI object to configure
+ * @param[in]     hz  The baud rate in Hz
+ */
+void spi_frequency(spi_t *obj, int hz)
+{
+    ADI_SPI_HANDLE  SPI_Handle;
+    ADI_SPI_RESULT  SPI_Return = ADI_SPI_SUCCESS;
+
+    SPI_Handle = *obj->pSPI_Handle;
+    SPI_Return = adi_spi_SetBitrate(SPI_Handle, (uint32_t) hz);
+    if (SPI_Return) {
+        obj->error = SPI_EVENT_ERROR;
+        return;
+    }
+}
+
+
+/** Write a byte out in master mode and receive a value
+ *
+ * @param[in] obj   The SPI peripheral to use for sending
+ * @param[in] value The value to send
+ * @return Returns the value received during send
+ */
+int spi_master_write(spi_t *obj, int value)
+{
+    ADI_SPI_TRANSCEIVER transceive;
+    uint8_t         TxBuf;
+    uint8_t         RxBuf;
+    ADI_SPI_HANDLE  SPI_Handle;
+    ADI_SPI_RESULT  SPI_Return = ADI_SPI_SUCCESS;
+
+    TxBuf = (uint8_t)value;
+
+    transceive.pReceiver        = &RxBuf;
+    transceive.ReceiverBytes    = 1;        /* link transceive data size to the remaining count */
+    transceive.nRxIncrement     = 1;        /* auto increment buffer */
+    transceive.pTransmitter     = &TxBuf;   /* initialize data attributes */
+    transceive.TransmitterBytes = 1;        /* link transceive data size to the remaining count */
+    transceive.nTxIncrement     = 1;        /* auto increment buffer */
+
+    transceive.bDMA = false;
+    transceive.bRD_CTL = false;
+    SPI_Handle = *obj->pSPI_Handle;
+    SPI_Return = adi_spi_MasterReadWrite(SPI_Handle, &transceive);
+    if (SPI_Return) {
+        obj->error = SPI_EVENT_ERROR;
+        return 1;
+    }
+
+    return((int)RxBuf);
+}
+
+
+/** Check if a value is available to read
+ *
+ * @param[in] obj The SPI peripheral to check
+ * @return non-zero if a value is available
+ */
+int spi_slave_receive(spi_t *obj)
+{
+    return 0;
+}
+
+
+/** Get a received value out of the SPI receive buffer in slave mode
+ *
+ * Blocks until a value is available
+ * @param[in] obj The SPI peripheral to read
+ * @return The value received
+ */
+int spi_slave_read(spi_t *obj)
+{
+    return 0;
+}
+
+
+/** Write a value to the SPI peripheral in slave mode
+ *
+ * Blocks until the SPI peripheral can be written to
+ * @param[in] obj   The SPI peripheral to write
+ * @param[in] value The value to write
+ */
+void spi_slave_write(spi_t *obj, int value)
+{
+    return;
```

Comment:
shouldnt this just not enabled SPI slave?

Reply:
The SPI slave mode is not implemented yet.

Reply:
It should not be in the codebase there. This is considered dead code, please add it once it is ready.